core/src/main/java/com/bumble/appyx/core/node/ParentNode.kt

```diff
@@ -284,6 +321,7 @@ abstract class ParentNode<Routing : Any>(
     }
 
     companion object {
+        const val AttachWorkFlowSyncTimeout = 1000L
```

Comment:
ALL_CAPS.
Also 1 seconds looks too fast, maybe 10?

Reply:
synchronisation of children with routing source happens on main thread, I think 1 second should be more than enough for the child to appear in the Parent's Node and we don't have to wait for additional 9 seconds in case it wasn't attached properly

Reply:
It's definitely possible for the main thread to be blocked for more than 1 second (ANRs)

Reply:
Waiting is suspended, no main thread blocks. IMHO 1 seconds still looks too fast, the main thread might be busy with something else, and wait function does not take it into account.

Reply:
Okay, I'll increase the timeout, but 10 is way too long in my opinion, should I make it 5 at least? So we'll fail faster

Reply:
@CherryPerry I was aware that this method wouldn't block the main thread, I was making the same point that you just made. The main thread could be busy for more than 1 second, causing this to time out.

For an ANR to occur, the UI thread is blocked for more than 5 seconds (more in some cases). There are quite a lot of apps that have more than 0% ANRs, so it is possible that this will time out. In this circumstance what would be the expectation from the developer?

The workflow would be cancelled? Is it then up to the developer to then code a fall-back mechanism for such a case, otherwise their app could be in an invalid state.

Perhaps there should be a version of this method that does not time out? (effectively the opposite problem to `waitForChildAttached`)

Reply:
I feel this error should definitely be instant. Even the 1 sec seems like a technicality. Increasing it further is bad ux (devx) imo. A random crash some time later, disconnected from the action itself. Can we instead get around it anyhow and get an instant crash?

Reply:
I made 1 sec by default with ability to override it. I don't like it as it think it shouldn't be part of the API but we can revise it later